components/gallery/List.vue

```diff
@@ -0,0 +1,303 @@
+<template>
+    <div class="gallery-list">
+        <ul class="list list-main">
+            <li
+                v-for="(main) in preparedList"
+                class="item-main"
+            >
+                <ul
+                    v-for="(list) in main"
+                    class="list"
+                >
+                    <li
+                        v-for="item in list.listItems"
+                        :key="item.id"
+                        class="item"
+                    >
+                        <h3
+                            class="item-title"
+                            @mouseout="mouseOut()"
+                            @mouseover="mouseOver(item)"
+                        >
+                            <nuxt-link
+                                :to="item.uri"
+                                class="link"
+                            >
+                                {{ item.title }}
+                            </nuxt-link>
+                        </h3>
+                    </li>
+                </ul>
+            </li>
+        </ul>
+
+        <transition
+            mode="out-in"
+            name="fade"
+        >
+            <wp-image
+                v-if="image"
+                :key="image.id"
+                :alt="image.altText"
+                :caption="caption"
+                :class="imageClasses"
+                :image="image"
+                mode="fullbleed"
+                @error-video="videoError"
+                @error-image="imageError"
+                @loaded-image="imageLoaded"
+                @loaded-video="videoLoaded"
+            />
+        </transition>
+    </div>
+</template>
+<script>
+
+import _dropRight from "lodash/dropRight"
+import _takeRight from "lodash/takeRight"
+
+export default {
+    name: "List",
+    props: {
+        items: {
+            type: Array,
+            default: () => []
+        },
+    },
+    data() {
+        return {
+            isImageLoaded: false,
+            isVideoLoaded: false,
+            isImageError: false,
+            isVideoError: false,
+            caption: '',
+            image: null
+        }
+    },
+    computed: {
+        imageClasses() {
+            return [
+                {
+                    'has-loaded-media': this.isImageLoaded || this.isVideoLoaded,
+                    'has-full-media-error': this.isVideoError && this.isImageError && !this.isImageLoaded && !this.isVideoLoaded,
```

Comment:
If you were just trying to show off classes, then cool.

But I think `wp-image` adds error and loading classes to itself already.

Reply:
WpImage adds the class `has-error` if there is a video error OR image error. So if the video fails to load but the image fallback does load WpImage still adds the class `had-error`. I needed to add a class based on whether or not one of the two media types loaded successfully `.has-loaded-media`. I also needed a way to add a class based on whether or not both the image and video fully errored `.has-full-media-error` and then show the caption if so.

Reply:
WpImage adds the class `has-error` if there is a video error OR image error. So if the video fails to load but the image fallback does load WpImage still adds the class `had-error`. I needed to add a class based on whether or not one of the two media types loaded successfully `.has-loaded-media`. I also needed a way to add a class based on whether or not both the image and video fully errored `.has-full-media-error` and then show the caption if so.